mmbench/workflow/predict.py

```diff
@@ -33,4 +43,97 @@ def benchmark_pred_exp(dataset, datasetdir, outdir):
     same number of samples and subjects, but possibly different latent
     dimensions.
     """
-    raise NotImplementedError
+    print_title("COMPARE MODELS USING REGRESSIONS "
+                f"OR CLASSIFICATION WITH ML ANALYSIS: {dataset}")
+    benchdir = outdir
+    print_text(f"Benchmark directory: {benchdir}")
+
+    print_subtitle("Loading data...")
+
+    latent_data = np.load(os.path.join(benchdir, f"latent_vecs_{dataset}.npz"))
+    latent_data_tr = np.load(os.path.join(benchdir,
+                                          f"latent_vecs_train_{dataset}.npz"))
+    assert latent_data.keys() == latent_data_tr.keys(), (
+        "latent data must have the same keys")
+    shape = None
+    meta_df = pd.read_csv(
+        os.path.join(benchdir, f"latent_meta_{dataset}.tsv"), sep="\t")
+    meta_df_tr = pd.read_csv(
+        os.path.join(benchdir, f"latent_meta_train_{dataset}.tsv"), sep="\t")
+    clinical_scores = meta_df_tr.columns
+    reg = linear_model.Ridge(alpha=.5)
+    cla = linear_model.RidgeClassifier()
+    predict_records, predict_results = dict(), dict()
+    for key in latent_data.keys():
+        samples = latent_data_tr[key]
+        samples_test = latent_data[key]
+        assert samples.ndim == 3 and samples_test.ndim == 3, (
+            "Expect samples with shape (n_samples, n_subjects, latent_dim).")
+        n_samples, n_subjects, _ = samples.shape
+        _, n_subjects_test, _ = samples_test.shape
+        if shape is None:
+            shape = samples.shape + (n_subjects_test,)
+        assert n_samples == shape[0] and samples_test.shape[0] == shape[0], (
+            "All samples must have the same number of samples.")
+        assert n_subjects == shape[1], (
+            "All samples must have the same number of subjects.")
+        assert n_subjects_test == shape[3], (
+            "All samples must have the same number of subjects for testing.")
```

Comment:
all test not necessary.
you can test if the train and test have the same latent space:

```suggestion
      assert samples.shape[-1] == samples_test.shape[-1], "The train and test data must be generated by the same model."
```